### PR TITLE
z_xxx_null functions added for all owned objects

### PR DESCRIPTION
**Pull request by milyin**
- added macro-defined z_xxx_null functions for all owned objects
- added test for safe drop of null objects
- fixed crashes on null objects dropping

The PR is based on https://github.com/eclipse-zenoh/zenoh-pico/pull/182

### CODE REVIEW

File: src/session/utils.c [hunk 1]
```diff
@@ -134,9 +134,8 @@ void _z_session_free(_z_session_t **zn) {
 }
 
 int8_t _z_session_close(_z_session_t *zn, uint8_t reason) {
-    int8_t ret = _Z_RES_OK;
-
-    ret = _z_transport_close(&zn->_tp, reason);
-
-    return ret;
+    if (zn == NULL) {
+        return _Z_ERR_GENERIC;
```
Comment (cguimaraes) [inline src/session/utils.c:138]: MISRA-C requires single point of return

File: src/net/primitives.c [hunk 2]
```diff
@@ -120,6 +120,10 @@ _z_publisher_t *_z_declare_publisher(_z_session_t *zn, _z_keyexpr_t keyexpr, z_c
 }
 
 int8_t _z_undeclare_publisher(_z_publisher_t *pub) {
+    if (pub == NULL) {
+        return _Z_ERR_GENERIC;
```
Comment (cguimaraes) [inline src/net/primitives.c:124]: MISRA-C requires single point of return

File: src/net/primitives.c [hunk 3]
```diff
@@ -175,25 +179,26 @@ _z_subscriber_t *_z_declare_subscriber(_z_session_t *zn, _z_keyexpr_t keyexpr, _
 }
 
 int8_t _z_undeclare_subscriber(_z_subscriber_t *sub) {
-    int8_t ret = _Z_RES_OK;
-
+    if (sub == NULL) {
+        return _Z_ERR_GENERIC;
```
Comment (cguimaraes) [inline src/net/primitives.c:183]: MISRA-C requires single point of return

File: src/net/primitives.c [hunk 4]
```diff
@@ -175,25 +179,26 @@ _z_subscriber_t *_z_declare_subscriber(_z_session_t *zn, _z_keyexpr_t keyexpr, _
...
-            // Only if message is successfully send, local subscription state can be removed
-            _z_unregister_subscription(sub->_zn, _Z_RESOURCE_IS_LOCAL, s);
-        } else {
-            ret = _Z_ERR_TRANSPORT_TX_FAILED;
-        }
-        _z_msg_clear(&z_msg);
-    } else {
-        ret = _Z_ERR_ENTITY_UNKNOWN;
+    if (s == NULL) {
+        return _Z_ERR_ENTITY_UNKNOWN;
```
Comment (cguimaraes) [inline src/net/primitives.c:187]: MISRA-C requires single point of return

**Review by cguimaraes — CHANGES_REQUESTED** [review]

File: src/session/utils.c (hunk 1 above)
Comment (cguimaraes) [reply src/session/utils.c]: Fixed.

File: src/net/primitives.c (hunk 2 above)
Comment (cguimaraes) [reply src/net/primitives.c]: Fixed.

File: src/net/primitives.c (hunk 3 above)
Comment (cguimaraes) [reply src/net/primitives.c]: Fixed.

File: src/net/primitives.c (hunk 4 above)
Comment (cguimaraes) [reply src/net/primitives.c]: Fixed.

File: src/net/primitives.c [hunk 5]
```diff
@@ -239,26 +244,25 @@ _z_queryable_t *_z_declare_queryable(_z_session_t *zn, _z_keyexpr_t keyexpr, _Bo
 }
 
 int8_t _z_undeclare_queryable(_z_queryable_t *qle) {
-    int8_t ret = _Z_RES_OK;
-
+    if (qle == NULL) {
+        return _Z_ERR_GENERIC;
```
[the remark this answers is not in the document]
Comment (cguimaraes) [reply src/net/primitives.c]: Fixed.

File: src/net/primitives.c [hunk 6]
```diff
@@ -239,26 +244,25 @@ _z_queryable_t *_z_declare_queryable(_z_session_t *zn, _z_keyexpr_t keyexpr, _Bo
...
-        if (_z_send_z_msg(qle->_zn, &z_msg, Z_RELIABILITY_RELIABLE, Z_CONGESTION_CONTROL_BLOCK) == _Z_RES_OK) {
-            // Only if message is successfully send, local queryable state can be removed
-            _z_unregister_questionable(qle->_zn, q);
-        } else {
-            ret = _Z_ERR_TRANSPORT_TX_FAILED;
-        }
-        _z_msg_clear(&z_msg);
-
+    if (q == NULL) {
+        return _Z_ERR_ENTITY_UNKNOWN;
```
[the remark this answers is not in the document]
Comment (cguimaraes) [reply src/net/primitives.c]: Fixed.

File: src/net/primitives.c (hunk 2 above)
Comment (milyin) [reply src/net/primitives.c]: Update looks good for me

**Review by cguimaraes — APPROVED** [review]
LGTM

**Pull request merged**